File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.kt

class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        var l = 0
        var r = numbers.size - 1
        while (l < r) {
            val sum = 
            if (numbers[l] + numbers[r] < target) {
                l++
            } else if (numbers[l] + numbers[r] > target) {
                r--
            }
            else {
                return intArrayOf(l + 1, r + 1)
            }
        }
        return intArrayOf()
    }
}

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.go
func twoSum(numbers []int, target int) []int {
    l := 0
    r := len(numbers) - 1

    for ;l < r; {
        if (target > numbers[l] + numbers[r]) {
            l++;
        } else if (target < numbers[l]+numbers[r]) {
            r--;
        } else {
            return []int{l + 1, r + 1};
        }
    }

    return nil
}